tests/server/test_resource_details.py

```diff
@@ -0,0 +1,327 @@
+"""
+    Copyright 2021 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import datetime
+from collections import defaultdict
+from typing import Dict
+from uuid import UUID
+
+import pytest
+
+from inmanta import data
+from inmanta.const import ResourceState
+from inmanta.data.model import ResourceVersionIdStr
+
+
+@pytest.fixture
+async def env_with_resources(server, client):
+    project = data.Project(name="test")
+    await project.insert()
+
+    env = data.Environment(name="dev", project=project.id, repo_url="", repo_branch="")
+    await env.insert()
+
+    env2 = data.Environment(name="dev2", project=project.id, repo_url="", repo_branch="")
+    await env2.insert()
+
+    env3 = data.Environment(name="dev3", project=project.id, repo_url="", repo_branch="")
+    await env3.insert()
+
+    cm_times = []
+    # Add multiple versions of model, with 2 of them released
+    for i in range(1, 6):
+        cm = data.ConfigurationModel(
+            environment=env.id,
+            version=i,
+            date=datetime.datetime.now(tz=datetime.timezone.utc),
+            total=1,
+            released=i != 1 and i != 5,
+            version_info={},
+        )
+        cm_times.append(cm.date)
+        await cm.insert()
+
+    cm = data.ConfigurationModel(
+        environment=env2.id,
+        version=4,
+        date=datetime.datetime.now(tz=datetime.timezone.utc),
+        total=1,
+        released=True,
+        version_info={},
+    )
+    cm_times.append(cm.date)
+    await cm.insert()
+
+    cm = data.ConfigurationModel(
+        environment=env3.id,
+        version=6,
+        date=datetime.datetime.now(tz=datetime.timezone.utc),
+        total=1,
+        released=True,
+        version_info={},
+    )
+    cm_times.append(cm.date)
+    await cm.insert()
+
+    async def create_resource(
+        path: str,
+        status: ResourceState,
+        version: int,
+        attributes: Dict[str, object],
+        agent: str = "internal",
+        resource_type: str = "std::File",
+        environment: UUID = env.id,
+    ):
+        key = f"{resource_type}[{agent},path={path}]"
+        res = data.Resource.new(
+            environment=environment,
+            resource_version_id=ResourceVersionIdStr(f"{key},v={version}"),
+            attributes={**attributes, **{"path": path}},
+            status=status,
+            last_deploy=datetime.datetime.now(),
```

Comment:
it might make the testcase easier to add some artificial timings, that are easy to check in the assertions?

(now, if all the times are sufficiently close, it will always pass)